src/dune/module_name.ml

```diff
@@ -41,24 +73,50 @@ let invalid_module_name ~loc name =
 let decode =
   let open Dune_lang.Decoder in
   plain_string (fun ~loc name ->
-      match name with
-      | "" -> invalid_module_name ~loc name
-      | s -> (
-        try
-          ( match s.[0] with
-          | 'A' .. 'Z'
-          | 'a' .. 'z' ->
-            ()
-          | _ -> raise_notrace Exit );
-          String.iter s ~f:(function
-            | 'A' .. 'Z'
-            | 'a' .. 'z'
-            | '0' .. '9'
-            | '\''
-            | '_' ->
-              ()
-            | _ -> raise_notrace Exit);
-          of_string s
-        with Exit -> invalid_module_name ~loc name ))
+      if is_valid_module_name name then
+        of_string name
+      else
+        invalid_module_name ~loc name)
 
 module Per_item = Per_item.Make (T)
+
+module Obj = struct
```

Comment:
Should we call this something like `Mangled`?
I just don't understand what object the name "Obj" refers to and this name doesn't communicate the "implementation detail" idea to me. (which, I presume, is the main idea the module is intended to communicate?)

Reply:
Mangled is a bit confusing as well since it's only mangled when necessary actually. Something like `Internal` or `Generated` could also possibly work. `Unique` is also pretty good at suggesting what the purpose of this name is.

Reply:
Speaking of unique, we mangle these to make them globally(ish) unique, but in the comment you'r saying unique within a library/executable. I'm improving the comment, but I'm not sure saying globally unique is correct either.

I think `Mangled` is a fine abbreviation for "possibly mangled", but I'm happy enough with `Unique`. I don't particularly like `Generated` because to me that suggests that the modules themselves might be generated.

Reply:
I guess I'll write "unique within a scope of a given linkage".